Persistent HTTP node list with connection status (#345)

tjpaulsondev opened the pull request:
I apologize if I missed something in the process. Looking to help how I can and this is an area that has impacted me so i though maybe i could help with this. it needs some more polish and testing but its not far off.

## Description

Adds persistent HTTP node connections with optional HTTPS toggle, enabling reconnect across page refresh or tab reload.

Introduces a unified tabbed UI for connecting via HTTP, Bluetooth, or Serial, reducing friction for new users and testing workflows.

## Related Issues

Relates to #345 
Fixes #241
Fixes #164 

## Changes Made

* Introduced new tabbed UI for connection modal (`HTTP`, `Bluetooth`, `Serial`)
* Implemented localStorage-based persistence of node connections
* Added HTTPS toggle and support for self-signed or trusted certs
* Updated HTTP connection logic for cleaner separation and recovery
* Restructured sections into reusable components for better clarity and extensibility
* Fixed numerous linting issues across touched files
* Minor cleanup of unused imports and unused variables (in progress)

## Testing Done

* Verified HTTP(S) node addition with:

  * Let's Encrypt–signed certs
  * Self-signed certs (via NGINX + local DNS)
* Confirmed nodes persist after refresh and page reloads
* Tested multi-node handling with different protocols
* Observed successful connection status returns in some cases

  * Needs further optimization for reliability
* Initial test results are promising, but status polling is incomplete
* Unencrypted HTTP (non-TLS) needs final testing and browser validation

## Screenshots (if applicable)

Connect Modal:
<img width="1224" alt="Updated connect modal with multiple persistent nodes" src="https://github.com/user-attachments/assets/959b6cc9-7985-4128-9140-e822091464c7" />
<img width="446" alt="Add new http connection" src="https://github.com/user-attachments/assets/e1c08d28-30f4-4e58-b77c-51f22d69d055" />

<img width="1219" alt="Bluetooth Tab" src="https://github.com/user-attachments/assets/da4d4aec-814b-422c-9b7d-ee0781df0052" />

<img width="1226" alt="Serial Tab" src="https://github.com/user-attachments/assets/9b1afddb-9092-4581-8bbb-7c32a72d1dcb" />

## Checklist

* [x] Code follows project style guidelines
* [ ] Documentation has been updated or added
* [ ] Tests have been added or updated
* [ ] All i18n translation labels have been added

## Future Enhancements

* Automatically reconnect to node if user refreshes while on the node’s detail screen
* Complete implementation for unencrypted HTTP node support
* Finalize connection status polling and error handling UX
* Integrate i18n labels in UI and confirm translation coverage

philon- committed:
Remove deprecated meshtastic/js dependency (#638)

* Remove deprecated meshtastic/js dependency

* Bump dependency version

* Fix linting

---------

Co-authored-by: philon- <philon-@users.noreply.github.com>

danditomaso reviewed (changes requested):
This is awesome work! Thank you for the contribution, I do see lots of files being committed with only formatting changes, so I'd like to ask you to ensure this work follows Deno's formatting rules. You can format this code by opening your branch and using "deno task format" to format the code accordingly.

danditomaso commented on src/components/ConnectionTabs/ConnectionTabs.tsx:
```diff
@@ -0,0 +1,58 @@
+  };
+
+  return (
+    <Tabs defaultValue="http" className={`w-full ${className || ""}`}>
```
On large monitors this would take up 2000px's for example. This width should be constrained to between  400-600 pxs (or whatever looks best)

danditomaso commented on src/components/Dashboard/SerialSection.tsx:
```diff
@@ -0,0 +1,233 @@
+            : (
+              <>
+                <Plus className="h-4 w-4 mr-2" />
+                Add Serial Device
```
These labels need to be added to our source translation files (which I know you called out in your PR summary), I know this process isn't documented currently. I'll have something available to reference today.

danditomaso commented on package-lock.json:
Please remove this file from your commit. Its causing the number of lines changed to become bloated because of it.

danditomaso commented on package.json:
```diff
@@ -36,10 +36,9 @@
-    "@meshtastic/transport-http": "npm:@jsr/meshtastic__transport-http",
-    "@meshtastic/transport-web-bluetooth": "npm:@jsr/meshtastic__transport-web-bluetooth",
-    "@meshtastic/transport-web-serial": "npm:@jsr/meshtastic__transport-web-serial",
+    "@meshtastic/transport-http": "npm:@jsr/meshtastic__transport-http@0.2.1",
```
Please revert this change, as a group we've left these speficially dependencies pinned to the latest versions

danditomaso commented on src/components/Dialog/Tabs/BluetoothTab.tsx:
This tabs folder should be moved under the Connect folder in PageComponents, since these components are used with each other they should live under the same parent folder.

danditomaso reviewed (changes requested):
Overall a great PR, I am seeing several things that need to be fixed prior to this being merged:
- Remove package.json from commit
- Use our built in formatting rules (use `deno task format`) to remove extraneous changes (most likely Biome auto formatted on save which caused these changes)
- You need to add the i18n translation labels to our source files, I'll make sure a markdown file is added on how to go about this.

tjpaulsondev committed:
Refactor connection tabs with persistent HTTP
  servers and i18n support

  - Move connection tabs from Dialog to PageComponents/Connect/Tabs for better
  organization
  - Add persistent HTTP server storage with status tracking in appStore
  - Implement hostname parsing and validation utilities
  - Default new HTTP connections to HTTPS for better security
  - Add comprehensive i18n translation support for all connection tab strings
  - Remove package-lock.json in favor of deno.lock dependency management
  - Update connection tab UI with improved server management (add/edit/remove)
  - Remove preliminary status checking logic (to be reimplemented with toradio API)
  - Maintain fixed window constraints for large monitor compatibility

tjpaulsondev committed:
Remove HTTP server status checking

  - Remove automatic server status checking from HTTP components
  - Eliminate status tracking overhead in HTTPServerSection and HTTP tab
  - Simplify persistent server list by removing status indicators
  - Clean up unused imports and status-related functions
  - Replace || with ?? for nullish coalescing consistency
